src/lib/route-builder.js

```diff
@@ -1,272 +0,0 @@
-      sourceAmount: quote.sourceAmount,
-      sourceHoldDuration: quote.sourceHoldDuration
-    }
-  }
-
-  /**
-   * Given a source transfer with an embedded final transfer, get the next
-   * transfer in the chain.
-   *
-   * It works as follows:
-   * Given `sourceTransfer` A→C, find the next hop B on the route from A to C.
-   * If the next hop is the final one (B == C), return the final transfer.
-   * Otherwise, return a transfer at B, with the final transfer C embedded.
-   *
-   * @param {Transfer} sourceTransfer
-   * @returns {Transfer} destinationTransfer
-   */
-  async getDestinationTransfer (sourceTransfer) {
-    log.info('constructing destination transfer ' +
-      'sourceLedger=%s sourceAmount=%s ilpPacket=%s',
-      sourceTransfer.ledger, sourceTransfer.amount, sourceTransfer.ilp)
-    if (!sourceTransfer.ilp) {
-      throw new IncomingTransferError(ilpErrors.F01_Invalid_Packet({
-        message: 'source transfer is missing "ilp"'
-      }))
-    }
-    let ilpPacket
-    try {
-      ilpPacket = packet.deserializeIlpPacket(Buffer.from(sourceTransfer.ilp, 'base64'))
-    } catch (err) {
-      log.debug('error parsing ILP packet: ' + sourceTransfer.ilp)
-      throw new IncomingTransferError(ilpErrors.F01_Invalid_Packet({
-        message: 'source transfer has invalid ILP packet'
-      }))
-    }
-    const destinationAddress = ilpPacket.data.account
-    const myAddress = this.ledgers.getPlugin(sourceTransfer.ledger).getAccount()
-    if (startsWith(destinationAddress, myAddress)) {
-      log.debug(
-        'ignoring transfer addressed to destination which starts with my address destination=%s me=%s',
-        destinationAddress,
-        myAddress
-      )
-      return
-    }
-
-    log.debug('constructing transfer for ILP packet with account=%s amount=%s',
-      ilpPacket.data.account, ilpPacket.data.amount)
-
-    const sourceLedger = sourceTransfer.ledger
-
-    // If the ILP packet is of type Forwarded Payment, it means we should forward the maximum
-    // we are willing to. Note that this feature is experimental, and support for it may disappear
-    // or change at any time.
-    const nextHop = (ilpPacket.type === packet.Type.TYPE_ILP_FORWARDED_PAYMENT)
-      ? await this.quoter.findBestPathForSourceAmount(sourceLedger, ilpPacket.data.account, sourceTransfer.amount)
-      : await this.quoter.findBestPathForFinalAmount(sourceLedger, ilpPacket.data.account, ilpPacket.data.amount)
-    if (!nextHop) {
-      log.info('could not find quote for source transfer: ' + JSON.stringify(sourceTransfer))
-      throw new IncomingTransferError(ilpErrors.F02_Unreachable({
-        message: 'No route found from: ' + sourceLedger + ' to: ' + ilpPacket.data.account
-      }))
-    }
-    this._verifyLedgerIsConnected(nextHop.destinationLedger)
-
-    // As long as the fxSpread > slippage, the connector won't lose money.
-    const expectedSourceAmount = new BigNumber(nextHop.sourceAmount).times(1 - this.slippage)
```

Comment:
This slippage computation wasn't added back in the refactor (the config option still exists, though).

Reply:
Removed.